ext/intl/grapheme/grapheme_util.c

```diff
@@ -126,6 +126,17 @@ int32_t grapheme_strpos_utf16(char *haystack, size_t haystack_len, char *needle,
 	ubrk_setText(bi, uhaystack, uhaystack_len, &status);
 	STRPOS_CHECK_STATUS(status, "Failed to set up iterator");
 
+	if (uneedle_len == 0) {
+		offset_pos = grapheme_get_haystack_offset(bi, last ? uhaystack_len : offset);
```

Comment:
As pointed out in the tests, just using uhaystack_len here is not right, as you may be doing a reverse scan with a negative offset, which has different semantics.

Reply:
So probably this should be `last && offset >= 0 ? uhaystack_len : offset`.

Reply:
Good catch, the grapheme_strrpos documentation doesn't specify negative offsets but it makes sense to bring it in line with mb_strrpos.